File: src/app/dtos/form/leave.dto.ts
export interface LeaveDTO {
  from_datetime : string;
  to_datetime : string;
  total_hours: number;
  type ?: String;
  reason : string;
  evident ?: string;
  approver_id: number;
  approver?: string;
  comment: string;
  status : string;
}

File: sys/src/main/java/com/hrms/sys/dtos/LeaveDTO.java
package com.hrms.sys.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;

import java.time.LocalDateTime;

@Data
@Builder
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class LeaveDTO {

    @JsonProperty("from_datetime")
    private LocalDateTime fromDatetime;

    @JsonProperty("to_datetime")
    private LocalDateTime toDatetime;

    @JsonProperty("type")
    private String type;

    @JsonProperty("reason")
    private String reason;

    @JsonProperty("total_hours")
    private Float totalHours;

    @JsonProperty("evident")
    private String evident;

    @JsonProperty("approver_id")
    private Long approverId;

    @JsonProperty("comment")
    private String comment;

    @JsonProperty("status")
    private String status;
}
